addons/account/models/account_payment_method.py

```diff
@@ -24,7 +24,8 @@ def create(self, vals_list):
         methods_info = self._get_payment_method_information()
         for method in payment_methods:
             information = methods_info.get(method.code)
-
+            if not information:
+                raise UserError(_("Invalid code. Only the following codes are allowed: %s") % ", ".join(methods_info.keys()))
```

Comment:
Hey, comma should be use for translation:
https://www.odoo.com/documentation/16.0/developer/howtos/translations.html#variables

Also, I am not sure I understand why/where we would let the user create payment method manually?

Reply:
Oops! thanks for the note :)

And yes, I think a better solution would be to hide the create button in the view. I haven't done it for not touching the operation... but if it's ok with you I make that change too.

Reply:
@Tardo why and where we would let the user create payment method manually from the UI ?

Reply:
@Tardo then I don't understand why we should apply this fix ?

Reply:
@tdetry If you are an administrator you can access the menu to create records of this type. This misleads users and gives a bad impression by displaying an "uncontrolled" error.

It is also strange to see that the field label says "(Do not modify)" but the field is not read-only.
Also the Spanish translation does not say anything about "(Do not modify)" showing only the text "Code".